### PR TITLE
fix: import panel exits on esc key (#4259)

### PR DESCRIPTION
**Pull request by vkuasar0**
- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added instructions to check how the fix works

Steps to check if the fix works:
1. Hover over a page in the side bar
2. Click on the + icon
3. Click on import, which causes a page to pop up
4. Now press the esc key to close the pop up

### CODE REVIEW

**Comment by vkuasar0** [comment]
Looks like CLAassisstant is not reading my GitHub account name properly. vis-kar was the older name of my account, and now it is vkuasar0. I have signed the Contributor License Agreement from this account.

**Comment by Xazin** [comment]
> Looks like CLAassisstant is not reading my GitHub account name properly. vis-kar was the older name of my account, and now it is vkuasar0. I have signed the Contributor License Agreement from this account.

It seems to be a problem with your local git configuration.

See more about this here: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

If your email is verified on GitHub, you should set that email in your local git configuration, see here how to do that: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git

---

Once you've done that, you will have to "fix" your commit, as it is committed with your old author, you should be able to simply run this command on your branch:
```
git commit --amend --author="First LastName <your-github-email>"
```
You will have to do a force push `git push --force` to replace the previous commit with the new commit.

File: frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart [hunk 1]
```diff
@@ -66,38 +67,7 @@ class ImportPanel extends StatelessWidget {
...
-                      FlowyOverlay.pop(context);
-                    }
-                  },
-                ),
-              ),
-            )
-            .toList(),
-      ),
-    );
+    return _buildKeyboardListeningFlowyContainer(context, height, width);
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart:70]: There's no reason to wrap the whole returning widget in a method. Just return the widget.

File: frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart [hunk 2]
```diff
@@ -162,6 +132,55 @@ class ImportPanel extends StatelessWidget {
...
+              )
+              .toList(),
+        ),
+      ),
+      onKeyEvent: (event) {
+        if (event is KeyDownEvent &&
+            event.physicalKey == PhysicalKeyboardKey.escape) {
+          FlowyOverlay.pop(context);
+        }
+      },
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart:181]: `child` properties should be listed last, put this at the top just below the `focusNode,` property

File: frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart [hunk 3]
```diff
@@ -162,6 +132,55 @@ class ImportPanel extends StatelessWidget {
...
   }
+
+  Widget _buildKeyboardListeningFlowyContainer(
+    BuildContext context,
+    double height,
+    double width,
+  ) {
+    return KeyboardListener(
+      autofocus: true,
+      focusNode: FocusNode(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/import/import_panel.dart:143]: FocusNode needs to be disposed, turn this widget into a StatefulWidget, and add a dispose method:
```dart
  final focusNode = FocusNode();

  @override
  void dispose() {
    focusNode.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    ...
```

**Comment by vkuasar0** [comment]
@Xazin
I configured git correctly, and made a force push after making the changes you recommended. Now ImportPanel is a StatefulWidget, and focus node is properly disposed.

**Comment by vkuasar0** [comment]
There was an unnecessary package import in a file, which was causing flutter build fail. I have edited it and force pushed the changes.

**Pull request merged**